plugins/modules/win_firewall_rule.ps1

```diff
@@ -166,7 +166,11 @@ try {
     if ($null -ne $enabled) { $new_rule.Enabled = $enabled } else { $new_rule.Enabled = $true }
     if ($null -ne $description) { $new_rule.Description = $description }
     if ($null -ne $group) { $new_rule.Grouping = $group }
-    if ($null -ne $program -and $program -ne "any") { $new_rule.ApplicationName = [System.Environment]::ExpandEnvironmentVariables($program) }
+    if ($null -ne $program -and $program -ne "any") { 
+        $new_rule.ApplicationName = [System.Environment]::ExpandEnvironmentVariables($program) 
+    } elseif ($program -eq "any") {
+        $new_rule.ApplicationName = $program
```

Comment:
This means that when someone sets `program: any` and a new rule is created the rule will have the program set `any` and not all like it does today. Where you trying to fix something by changing this logic?

Reply:
Yes, `Set-NetFirewallRule` accepts `any` for the program parameter and nulls the property. I won't be neccessary anymore when applying your solution.